grails-app/controllers/org/pih/warehouse/core/DocumentController.groovy

```diff
@@ -257,6 +265,10 @@ class DocumentController {
         } else {
             log.info "Document is too large"
             flash.message = "${warehouse.message(code: 'document.documentTooLarge.message')}"
+            if (stockMovement) {
+                redirect(controller: 'stockMovement', action: 'show', id: stockMovement.id)
+                return
+            }
             if (shipmentInstance) {
                 redirect(controller: 'stockMovement', action: 'show', id: command.shipmentId)
                 return
```

Comment:
So if you are working on it, can you remove those disgusting elses when there are returns already?

Reply:
honestly I would prefer not to refactor this causing a headache code right now, just before the release, as this is a generic method that I would have to test for many workflows (SMs, invoices, etc.).

Reply:
I agree. Like my previous comment says, this class definitely could use a refactor but now is probably not the best time.